wgpu-core/src/device/resource.rs

```diff
@@ -359,6 +358,14 @@ impl<A: HalApi> Device<A> {
         }
     }
 
+    pub fn get_queue(&self) -> Option<Arc<Queue<A>>> {
+        self.queue.read().as_ref()?.upgrade()
+    }
+
+    pub fn set_queue(&self, queue: Arc<Queue<A>>) {
+        self.queue.write().replace(Arc::downgrade(&queue));
```

Comment:
I feel like there should be an assert that this is None - this function should never be called more than once. In fact we could use a OnceCell for the Arc instead of an Mutex<Option<>> - that more actually models what we want (late init)